dashboard/app/models/school_info.rb

```diff
@@ -159,12 +149,30 @@ def validate_non_us
   end
 
   def validate_us
+    # The right side of this expression should not be necessary, but there are a ton
+    # of legacy tests in school_info_test.rb that will fail without it. Rather than deleting
+    # the old tests, we decided to keep them until all of the front end components that update
+    # school_info are refactored to send the appropriate data. See "Validate records" comment
+    # above to understand the required data for nces and non-nces schools
+    if school || (school_type && school_type != SCHOOL_TYPE_NO_SCHOOL_SETTING)
+      validate_nces_school
+    else
+      validate_non_nces_school
+    end
+  end
+
+  def validate_nces_school
     errors.add(:school_type, "is required") unless school_type
     errors.add(:school_type, "is invalid") unless SCHOOL_TYPES.include? school_type
     validate_private_other if [SCHOOL_TYPE_PRIVATE, SCHOOL_TYPE_OTHER].include? school_type
     validate_public_charter if [SCHOOL_TYPE_PUBLIC, SCHOOL_TYPE_CHARTER].include? school_type
   end
 
+  def validate_non_nces_school
+    errors.add(:zip, "is required") unless zip
+    errors.add(:school_name, "is required") unless school_name || school_type == SCHOOL_TYPE_NO_SCHOOL_SETTING
+  end
+
   def validate_private_other
```

Comment:
Do we have 'private other' schools anymore? Or are all schools either 'nces' or 'non-nces'?

Reply:
good question! I think we mostly don't need this anymore. `private`, `public`, and `charter` school types are only known if the school is an nces school, we no longer ask the user for school type if their school is not in the list. `other` school type is basically deprecated at this point. the `validation_private_other` function is essentially just ensuring a private nces school doesn't have an associated school district 🤷 

I think there's still a fair amount of clean up to be done here, but since I feel like I learn something new about school_infos with each task, I'm erring on the side of caution by preserving some of the legacy code until everything is fully switched over